### PR TITLE
Rpr 45

### PR DESCRIPTION
**Pull request by linorabolini**
Hey guys, 

Just continuing the work ! 

This PR has all previous changes ( RPR-50 / RPR-48 ) plus a simplified version of the renderer flow.
This tracer code would be complete and ads get rendered in the WP page when using the generated script.

Further steps are to develop the missing plugins so we can mimic the same amount of features that the other version has, but the main skeleton/architecture would be defined in this PR.

We need to sync to see how the configuration would be injected into the code, for now I'm just using the WP window globals that we have in prod

### CODE REVIEW

File: src/core/renderer/index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        // to store all nodes that will be render
+        // in the current cycle
+        adNodesToRender: [],
+
+        // to allow googletag refresh to be delayed
+        renderCallDelay: 0
+    })
+
+    // init lifecycle
```
Comment (eric-washpost) [inline src/core/renderer/index.js]: If we are going to attach more features on the different phases for the lifecycle, we need to attach the function calls on here? If not, we attach on the different files, but we may lost tracking how many events or features are attached on the lifecycle. We may need to manage these function calls on one JS file?

File: src/utils/index.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import eventify from "./eventify"
+
+function getCookie(name) {
+    var cookie = "" + document.cookie,
```
Comment (eric-washpost) [inline src/utils/index.js]: var?? why not let?

File: src/utils/index.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    const regex =
+        type === "variable"
+            ? new RegExp(`[\\?&;]${arg}=([^&#?]*)`)
+            : new RegExp(`[\\?&;](${arg})[^=&#?]*`)
+    const results = regex.exec(url)
+    return results === null ? null : results[results.length - 1]
+}
+
+function addInlineCss(str) {
+    var el = document.createElement("style")
```
Comment (eric-washpost) [inline src/utils/index.js]: let?

File: src/utils/index.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    if (el.styleSheet) {
+        el.styleSheet.cssText = str
+    } else {
+        el.appendChild(document.createTextNode(str))
+    }
+    return document.getElementsByTagName("head")[0].appendChild(el)
+}
+function setTargeting(map, obj) {
+    for (var key in map) {
+        if (map.hasOwnProperty(key)) {
```
Comment (eric-washpost) [inline src/utils/index.js]: do we have a better way to loop object? `for (let key of Object.keys(obj))` is better?

File: src/utils/index.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            obj.setTargeting(key, map[key])
+        }
+    }
+}
+function setCategoryExclusion(map, obj) {
+    for (var key in map) {
```
Comment (eric-washpost) [inline src/utils/index.js]: see above? looping object

File: src/utils/index.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+            regex.push(wordList[l] + variations)
+        }
+        regex = "\\b" + regex.join("\\b|\\b") + "\\b"
+        return new RegExp(regex, "i").test(str)
+    }
+    return false
+}
+
+function loadScripts(scripts) {
+    var script,
```
Comment (eric-washpost) [inline src/utils/index.js]: const, let?

File: src/index.js [hunk 7]
```diff
@@ -1,6 +1,53 @@
...
+import targeting from "core/plugins/targeting"
+import KeyValues from "keyvalues"
+import tmpKV from "./tmpKV"
 
-const kv = new KeyValues();
+// keyvalues for targeting
 
-(() => alert(kv.run({ test: "it works" })))()
+let kv = new KeyValues()
+kv.addPlugin(tmpKV())
```
Comment (eric-washpost) [inline src/index.js]: I'm assuming the keyvalues are going to rewrite?

Comment (linorabolini) [reply src/index.js]: yes this is just temp: https://arcpublishing.atlassian.net/browse/RPR-40?filter=-1&jql=assignee%20%3D%20currentUser()%20AND%20resolution%20%3D%20Unresolved%20ORDER%20BY%20priority%20DESC%2C%20cf%5B11102%5D%20DESC%2C%20updated%20DESC 

:)

**Comment by eric-washpost** [comment]
Merge it to the dev for the initial project commit. We will fix issues on coming commits.

**Pull request merged**